test/src/test/java/hudson/model/ComputerTest.java

```diff
@@ -263,4 +266,21 @@ public void testTerminatedNodeAjaxExecutorsDoesNotShowTrace() throws Exception {
         j.assertBuildStatus(Result.FAILURE, j.waitForCompletion(b));
     }
 
+    @Test
+    public void computersCollected() throws Exception {
+        DumbSlave agent = j.createOnlineSlave();
+        FreeStyleProject p = j.createFreeStyleProject();
+        p.setAssignedNode(agent);
+        j.buildAndAssertSuccess(p);
+        Computer computer = agent.toComputer();
+        WeakReference<Computer> computerRef = new WeakReference<>(computer);
+        WeakReference<Channel> channelRef = new WeakReference<>((Channel) computer.getChannel());
```

Comment:
Just in case there was some problem with the `Channel` separately from the `Computer`. I could not reproduce any in this test, other than https://github.com/jenkinsci/remoting/pull/694 which was reproduced interactively.